code/modules/mob/living/silicon/subsystems.dm

```diff
@@ -33,6 +41,10 @@
 	law_manager		= new(src)
 	power_monitor	= new(src)
 
+/mob/living/silicon/robot/init_subsystems()
+	. = ..()
+	self_diagnosis  = new(src)
```

Comment:
I don't believe this is cleaned up on `Destroy()`, so it might cause problems with GC but I'm not sure

Reply:
You're probably right. I'll look into it and qdel it if need be.

Reply:
Well, it turns out `/mob/living/silicon/Destroy()` doesn't even qdel these, so I guess it's just handled with a hard del. Tempted to do a `QDEL_NULL` on all subsystems.